### PR TITLE
Import jstz.min.js from jstimezonedetect 1.0.7

### LINKED ISSUE #543: "Uncaught ReferenceError: jstz is not defined" error in browser console
### Steps to reproduce
1. Install NC 22.1.1 (still happening with NC 22.2.3)
2. Log in through user_saml app
3. Check browser console

### Expected behaviour
No javascript error in the browser console

### Actual behaviour
This error is displayed in the browser console:
_Uncaught ReferenceError: jstz is not defined_
related to file _/apps/user_saml/js/timezone.js_ , line 30

### Server configuration
**Operating system**: Ubuntu 18

**Web server:** Apache

**PHP version:** 7.4

**Nextcloud version:** 22.1.1

**Where did you install Nextcloud from:** Download of zip file from https://download.nextcloud.com/server/releases

### Client configuration
**Browser:** Firefox 78.13

**Operating system:** Windows 10

#### Browser log
Uncaught ReferenceError: jstz is not defined
    <anonymous> https://server_url/apps/user_saml/js/timezone.js?v=fc1b204a-5:30

I think it is related to that PR which removed jstz:
https://github.com/nextcloud/server/pull/25850

### PR DESCRIPTION
**Pull request by danxuliu**
Fixes #543

[In Nextcloud 22 `jstimezonedetect` was removed from the server bundle](https://github.com/nextcloud/server/pull/25850), so now each app that uses it needs to ship its own copy. For simplicity, as the app does not currently use webpack or NPM, the minified file was just added to _js/vendor_.

### CODE REVIEW

**Comment by blizzz** [comment]
@danxuliu here, master serves Nextcloud 21to 24. Would the change be a problem on 21?

**Comment by danxuliu** [comment]
> @danxuliu here, master serves Nextcloud 21 to 24. Would the change be a problem on 21?

In Nextcloud 21 `jstimezonedetect` was shipped in the server bundle, but [it was already deprecated](https://github.com/nextcloud/server/blob/v21.0.7/core/src/globals.js#L110). Deprecated global properties are [added to `window` as a read-only function that prints a warning and then returns the wrapped property](https://github.com/nextcloud/server/blob/v21.0.7/core/src/globals.js#L88-L98). As it is read-only trying to [set `window.jstz`](https://github.com/pellepim/jstimezonedetect/blob/ddc9e040342d68c06129ae28011517e5b2dde664/dist/jstz.js#L1457) later when _vendor/jstz.min.js_ is loaded by _user_saml_ has no effect ([not even a console error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#modifying_a_property), as _vendor/jstz.min.js_ is not in strict mode).

In short, no, the change should not be a problem in Nextcloud 21 :-)

**Review by blizzz — APPROVED** [review]

**Review by acsfer — APPROVED** [review]

**Comment by HolgerHees** [comment]
Hi, any progress here? pull request is still blocked because of an "fatal: refusing to merge unrelated histories" error. Maybe a rebase would help?

**Comment by HolgerHees** [comment]
looks like the integration tests itself are broken.

This version of Nextcloud requires at least PHP 7.4<br/>You are currently running 7.3.20. Please update your PHP version.This version of Nextcloud requires at least PHP 7.4<br/>You are currently running 7.3.20. Please update your PHP version.

Seams to be not related to your pull request.

anyone has an idea who is able to fix and where we should report this?

**Comment by danxuliu** [comment]
> anyone has an idea who is able to fix and where we should report this?

See https://github.com/nextcloud/user_saml/pull/583 :-)
